x-pack/plugins/uptime/server/lib/requests/get_last_successful_step.ts

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { UMElasticsearchQueryFn } from '../adapters/framework';
+import { Ping } from '../../../common/runtime_types/ping';
+
+export interface GetStepScreenshotParams {
+  monitorId: string;
+  timestamp: string;
+  stepIndex: number;
+}
+
+export const getStepLastSuccessfulStep: UMElasticsearchQueryFn<
+  GetStepScreenshotParams,
+  any
+> = async ({ uptimeEsClient, monitorId, stepIndex, timestamp }) => {
+  const lastSuccessCheckParams = {
+    size: 1,
+    sort: [
+      {
+        '@timestamp': {
+          order: 'desc',
+        },
+      },
+    ],
+    query: {
+      bool: {
+        filter: [
+          {
+            range: {
+              '@timestamp': {
+                lte: timestamp,
+              },
+            },
+          },
+          {
+            term: {
+              'monitor.id': monitorId,
+            },
+          },
+          {
+            term: {
+              'synthetics.type': 'step/end',
+            },
+          },
+          {
+            term: {
+              'synthetics.payload.status': 'succeeded',
```

Comment:
This shouldn't work because we use `enabled: false` in our mapping for `synthetics.payload`. If it is working on your box I'd assume it's a broken mapping.

If we need to filter in this way we should add `synthetics.step.status` as a field. I've opened https://github.com/elastic/beats/issues/24141 to track this

Reply:
agreed, our mapping was broken, only @paulb-elastic had it right all along.